### PR TITLE
[_]: Feat/send notification on user storage changed

### PR DESCRIPTION
**Pull request by fabioespinosa**
Uses the event bus to emit a an event which triggers a notification showing that a user has changed storage

### CODE REVIEW

File: lib/engine.js [hunk 1]
```diff
@@ -92,11 +101,14 @@ Engine.prototype.start = function (callback) {
...
-    log.error('error connecting to worker queue: %s', err.message);
-  });
+  this.networkQueue
+    .connectAndRetry()
+    .then(() => {
+      log.info('connected to worker queue');
+    })
+    .catch((err) => {
+      log.error('error connecting to worker queue: %s', err.message);
+    });
```
Comment (fabioespinosa) [inline lib/engine.js:112]: Above is prettier (ignore)

File: lib/engine.js [hunk 2]
```diff
@@ -178,7 +190,9 @@ Engine.prototype._configureApp = function () {
       const corsRegex = new RegExp(self._config.server.corsRegex);
       const originAllowed = function () {
         if (process.env.NODE_ENV !== 'production') {
-          log.warn('Insecure CORS configuration in use; non-production env detected');
+          log.warn(
+            'Insecure CORS configuration in use; non-production env detected'
+          );
```
Comment (fabioespinosa) [inline lib/engine.js:196]: Above is prettier (ignore)

File: lib/engine.js [hunk 3]
```diff
@@ -210,13 +224,21 @@ Engine.prototype._configureApp = function () {
...
     res.on('close', () => {
-      const durationInMilliseconds = getDurationInMilliseconds (start);
-      log.info(`${req.method} ${req.originalUrl} [CLOSED] ${durationInMilliseconds.toLocaleString()} ms`);
+      const durationInMilliseconds = getDurationInMilliseconds(start);
+      log.info(
+        `${req.method} ${
+          req.originalUrl
+        } [CLOSED] ${durationInMilliseconds.toLocaleString()} ms`
+      );
     });
```
Comment (fabioespinosa) [inline lib/engine.js:242]: Above is prettier (ignore)

File: lib/engine.js [hunk 4]
```diff
@@ -233,11 +255,12 @@ Engine.prototype._configureApp = function () {
...
-  const protocol = this._config.server.ssl &&
+  const protocol =
+    this._config.server.ssl &&
     this._config.server.ssl.cert &&
-    this._config.server.ssl.key ?
-    'https:':
-    'http:';
+    this._config.server.ssl.key
+      ? 'https:'
+      : 'http:';
```
Comment (fabioespinosa) [inline lib/engine.js:264]: Above is prettier (ignore)

File: lib/engine.js [hunk 5]
```diff
@@ -312,41 +336,43 @@ Engine.prototype._countPendingResponses = function () {
...
+    {
+      connections: (next) => {
+        this.server.server.getConnections(next);
+      },
+    },
+    (err, results) => {
+      if (err) {
+        // Error message is included in results below
+        results = {};
+      }
```
Comment (fabioespinosa) [inline lib/engine.js:350]: Above is prettier (ignore)

File: lib/engine.js [hunk 6]
```diff
@@ -312,41 +336,43 @@ Engine.prototype._countPendingResponses = function () {
...
+        memory: process.memoryUsage(),
+        heapStatistics: v8.getHeapStatistics(),
+        heapSpaceStatistics: v8.getHeapSpaceStatistics(),
+        uptime: process.uptime(),
+        listening: this.server.server.listening,
+        connections: results.connections,
+        pendingResponses: this._countPendingResponses(),
+        databaseState: this.storage.connection.readyState,
+        redisConnected: this.redis.connected,
+      };
```
Comment (fabioespinosa) [inline lib/engine.js:368]: Above is prettier (ignore)

**Review by sg-gs — APPROVED** [review]

**Pull request merged**